### ISSUE TITLE
Crash on start - 'ascii' codec can't decode byte 0xe2 in position 1189: ordinal not in range(128)

### ISSUE THREAD

**Issue by Penaz91** [The-Compiler/qutebrowser#1151]
First of all, thank you for bringing this amazing browser in my life, i was used to DWB and i found a great and far more stable alternative in qutebrowser.
After a full system update, qutebrowser crashes with the following traceback:

```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.4/qutebrowser", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 3143, in <module>
    @_call_aside
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 3129, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 3156, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 649, in _build_master
    ws = cls()
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 642, in __init__
    self.add_entry(entry)
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 698, in add_entry
    for dist in find_distributions(entry, True):
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 2136, in find_on_path
    path_item, entry, metadata, precedence=DEVELOP_DIST
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 2518, in from_location
    version = cls._version_from_metadata(dist_path) or version
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 2841, in _version_from_metadata
    return _version_from_file(strm)
  File "/usr/lib64/python3.4/site-packages/pkg_resources/__init__.py", line 2483, in _version_from_file
    line = next(iter(version_lines), '')
  File "/usr/lib64/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1189: ordinal not in range(128)
```

I can't manage to get any further logs on my own at the moment. Sorry.
I'll attach the symbols as soon as possible.

**Comment by Penaz91** [comment]
Sadly, when i try to get the symbols, gdb just creates a file with "No Stack" written inside.

**Comment by The-Compiler** [comment]
This is a Python exception rather than a C++ crash, so no need for symbols or debugging with `gdb`. I'm not sure what's going on though, I'll investigate.

**Comment by Penaz91** [comment]
This might help:
I'm on Gentoo Linux and I recently updated from setuptools-18.6.1 to setuptools-18.7.
Reverting to the previous version solves the issue.

**Comment by The-Compiler** [comment]
How did you install qutebrowser? I just did the same update on my Archlinux without any issues.

I'm not sure what file it's trying to read there either - `0xe2` is invalid UTF-8 and `â` in latin1, neither of which should be anywhere in qutebrowser's files.

Maybe this should be reported to https://bitbucket.org/pypa/setuptools/issues instead? (Please link the issue if you do)

**Comment by The-Compiler** [comment]
Maybe a `grep -R 'â' /usr/lib64/python3.4/site-packages/*.egg-info` reveals what it's choking on?

**Comment by Penaz91** [comment]
I installed qutebrowser 0.4.1-r1 via Portage.
I don't think that kind of grep is possible in my console, as the symbol gets shown as a question mark and using "\u00e2" "\x00e2" in place of the symbol said returns nothing.

**Comment by The-Compiler** [comment]
maybe try `grep -R $(echo -e '\xe2') /usr/lib64/python3.4/site-packages/*.egg-info`

**Comment by Penaz91** [comment]
This is what i got:

/usr/lib64/python3.4/site-packages/characteristic-14.3.0-py3.4.egg-info/PKG-INFO:        ``characteristic``\ ’s documentation lives at `Read the Docs <https://characteristic.readthedocs.org/>`_, the code on `GitHub <https://github.com/hynek/characteristic>`_.
/usr/lib64/python3.4/site-packages/characteristic-14.3.0-py3.4.egg-info/PKG-INFO:        It’s rigorously tested on Python 2.6, 2.7, 3.3+, and PyPy.
/usr/lib64/python3.4/site-packages/characteristic-14.3.0-py3.4.egg-info/PKG-INFO:        It’s inspired by Twisted’s `FancyEqMixin <http://twistedmatrix.com/documents/current/api/twisted.python.util.FancyEqMixin.html>`_ but is implemented using class decorators because `sub-classing is bad for you <https://www.youtube.com/watch?v=3MNVP9-hglc>`_, m’kay?
/usr/lib64/python3.4/site-packages/idna-2.0-py3.4.egg-info/PKG-INFO:        This version of the protocol is often referred to as “IDNA2008” and can 
/usr/lib64/python3.4/site-packages/idna-2.0-py3.4.egg-info/PKG-INFO:        the “encodings.idna” module that comes with the Python standard library
/usr/lib64/python3.4/site-packages/pexpect-4.0.1-py3.4.egg-info:        library, which is only available on Unix-like systems. Some features—waiting
/usr/lib64/python3.4/site-packages/pexpect-4.0.1-py3.4.egg-info:        for patterns from file descriptors or subprocesses—are also available on
/usr/lib64/python3.4/site-packages/service_identity-14.0.0-py3.4.egg-info/PKG-INFO:        **TL;DR**: Use this package if you use pyOpenSSL_ and don’t want to be MITM_\ ed.
/usr/lib64/python3.4/site-packages/service_identity-14.0.0-py3.4.egg-info/PKG-INFO:        ``service_identity``\ ’s documentation lives at `Read the Docs <https://service-identity.readthedocs.org/>`_, the code on `GitHub <https://github.com/pyca/service_identity>`_.

**Comment by The-Compiler** [comment]
Fun. Seems like that `’` ([U+2019](http://www.fileformat.info/info/unicode/char/2019/index.htm)) is `0xE2 0x80 0x99` in UTF-8 and `setuptools` is choking on that - this is definitely unrelated to qutebrowser though.

I bet if you do `grep -R load_entry_point /usr/bin` and try to run anything you get there, you'll get the ssame issue.

If I'm right, can you please report that bug to setuptools? Let me know if you'd prefer me to do it.

I also *think* this should not happen if your locale is configured properly. I'm guessing `locale` will tell you your locale is `C`, which means your system can't handle non-ASCII data. It's probably a good idea to fix that on your system too, if that's the case.

**Comment by Penaz91** [comment]
You got it partially, I ran some programs from the list i got from your grep command, but all of them worked.
But i tried setting my locale from "C" to "en_US.UTF-8" and Qutebrowser seems to be running successfully.
In console i noticed this traceback though:
WARNING: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-penaz'
  File "/usr/lib/python-exec/python3.4/qutebrowser", line 9, in <module>
    load_entry_point('qutebrowser==0.4.1', 'gui_scripts', 'qutebrowser')()
  File "/usr/lib64/python3.4/site-packages/qutebrowser/qutebrowser.py", line 149, in main
    return app.run(args)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/app.py", line 96, in run
    server = ipc.send_or_listen(args)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/misc/ipc.py", line 491, in send_or_listen
    socketname = _get_socketname(args.basedir)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/misc/ipc.py", line 67, in _get_socketname
    target_dir = standarddir.runtime()
  File "/usr/lib64/python3.4/site-packages/qutebrowser/utils/standarddir.py", line 56, in runtime
    return _get(QStandardPaths.RuntimeLocation)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/utils/standarddir.py", line 134, in _get
    path = _writable_location(typ)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/utils/standarddir.py", line 67, in _writable_location
    path = QStandardPaths.writableLocation(typ)
  File "/usr/lib64/python3.4/site-packages/qutebrowser/utils/log.py", line 323, in qt_message_handler
    stack = ''.join(traceback.format_stack())

Anyway, the browser seems to be back to its usual awesomeness. Thank you a lot for your time.

**Comment by The-Compiler** [comment]
I also reported this as https://bitbucket.org/pypa/setuptools/issues/469/ now.

The warning is due to Qt following the [XDG basedir specification](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) and your system not setting `XDG_RUNTIME_DIR` - I'm still unsure if anything other than systemd does that though. I'll probably hide it in the future, but I'm not sure yet.

**Comment by The-Compiler** [comment]
FYI, this has been fixed in setuptools 18.7.1.

**Comment by Penaz91** [comment]
Thanks a ton for the information, i'll update as soon as I can, even if the locale fix is applied.
